src/Commands/TankDrive.cpp

```diff
@@ -42,6 +42,32 @@ void TankDrive::Execute() {
 	RobotMap::chassisrightMotor2->Set(rY);
 	RobotMap::chassisrightMotor3->Set(rY);
 
+	//Automatic Shifters
+	SmartDashboard::PutNumber("EncodeValue chassisrightmotor2",
+			RobotMap::chassisrightMotor2->GetEncVel());
+	double shiftUpVelocity = SmartDashboard::GetNumber("UpVelocity", 3000);
```

Comment:
Are you trying to get a number from preferences? This gets a number from the SmartDashboard, nothing to do with Robot Preferences. To get numbers of Preferences use `Preferences::GetInstances()->GetDouble("key", defaultValue) ` this returns a number with the key. You can save an instance of the Preferences if you want. Would be something like `Preferences* prefs = Preferences::GetInstance()` then call `prefs->GetDouble("key", defaultValue)`